opentelemetry-api/src/opentelemetry/configuration/__init__.py

```diff
@@ -0,0 +1,109 @@
+# Copyright 2020, OpenTelemetry Authors
```

Comment:
can documentation be added around how the Configuration module works, or rather configuration in general?

Reading through this code the behavior is somewhat clear to me, but I think users will need a bit more handholding.

Reply:
Yes, definitely, adding more documentation right now :+1: